packages/admin/admin/src/form/FinalFormFileSelect.tsx

```diff
@@ -0,0 +1,273 @@
+    disableSelectFileButton,
+    accept,
+    maxSize = 50 * 1024 * 1024,
+    maxFiles,
+    input: { onChange, value: fieldValue, multiple: multipleFiles },
+    iconMapping = {},
+}) => {
+    const { delete: deleteIcon = <Delete />, info: infoIcon = <Error color="error" />, select: selectIcon = <Select /> } = iconMapping;
+
+    const onDrop = React.useCallback(
+        (acceptedFiles: File[]) => {
+            if (multipleFiles) {
+                if (Array.isArray(fieldValue)) {
+                    if (!maxFiles || (maxFiles && fieldValue.length < maxFiles && fieldValue.length + acceptedFiles.length <= maxFiles)) {
+                        onChange([...fieldValue, ...acceptedFiles]);
+                    }
+                } else {
+                    onChange([...acceptedFiles]);
+                }
+            } else {
+                onChange(acceptedFiles[0]);
+            }
+        },
+        [fieldValue, multipleFiles, onChange, maxFiles],
+    );
+
+    const removeFile = (removedFile: File) => () => {
+        const newFiles = Array.isArray(fieldValue) ? fieldValue.filter((file) => file !== removedFile) : undefined;
+        onChange(newFiles);
+    };
+
+    const { fileRejections, getRootProps, getInputProps, isDragReject } = useDropzone({
+        onDrop,
+        accept,
+        disabled: disabled || (maxFiles && fieldValue.length >= maxFiles),
+        multiple: multipleFiles,
+        maxSize: maxSize,
+        maxFiles,
+    });
+
+    let acceptedFiles: File[] = [];
+
+    if (Array.isArray(fieldValue)) {
+        acceptedFiles = fieldValue;
+    } else if (fieldValue.name !== undefined) {
+        acceptedFiles = [fieldValue];
+    }
+
+    const rejectedFiles = fileRejections.map((rejectedFile, index) => (
+        <div key={index} className={classes.rejectedFileListItem}>
+            <Tooltip trigger="hover" title={rejectedFile.file.name}>
+                <div className={classes.fileListText}>{rejectedFile.file.name}</div>
+            </Tooltip>
+            {infoIcon}
+        </div>
+    ));
+
+    return (
+        <div className={classes.root}>
+            <div {...getRootProps()} className={classes.dropzone}>
+                <input {...getInputProps()} />
+                {!disableDropzone && (
+                    <div
+                        className={clsx(
+                            classes.droppableArea,
+                            disabled && classes.droppableAreaIsDisabled,
+                            isDragReject && classes.droppableAreaHasError,
+                        )}
+                    >
+                        {isDragReject && <div className={classes.droppableAreaError}>{infoIcon}</div>}
+                        <Typography variant="body2" className={classes.droppableAreaCaption}>
+                            <FormattedMessage id="comet.finalFormFileSelect.dropfiles" defaultMessage="Drop files here to upload" />
+                        </Typography>
+                    </div>
```

Comment:
When I have uploaded the maximum amount of files, there is no real indication, that I cannot upload more, it just stops working.  
Should the `droppableArea` maybe be hidden? 
Should there be some kind of message to inform the user, that they cannot upload more files?

https://github.com/vivid-planet/comet/assets/6264317/ff1dccbb-970a-4675-b7d9-4ca60ac0ced8

Reply:
It's easier to see if Dropzone is disabled with the Button:
<img width="740" alt="Screenshot 2024-05-02 at 13 49 30" src="https://github.com/vivid-planet/comet/assets/72875628/8cbe7619-0110-43bf-bb10-0cc74e540c00">
I added a maximum files amount message after the Dropzone, does it help?

Reply:
The dropzone is still present and clearly states that the user can upload files, even though that is not possible. 
It should be obvious to the user, that they cannot upload more files. Maybe this should be discussed with the UX team.